packages/volto/src/helpers/Url/Url.js

```diff
@@ -94,14 +100,30 @@ export function getView(url) {
  * @returns {string} Flattened URL to the app server
  */
 export function flattenToAppURL(url) {
+  let adjustedUrl = url;
   const { settings } = config;
-  return (
-    url &&
-    url
+  const prefix = settings.prefixPath;
+
+  adjustedUrl =
+    adjustedUrl &&
+    adjustedUrl
       .replace(settings.internalApiPath, '')
       .replace(settings.apiPath, '')
-      .replace(settings.publicURL, '')
-  );
+      .replace(settings.publicURL, '');
+  const internalURL =
+    adjustedUrl &&
+    (adjustedUrl.indexOf(settings.publicURL) !== -1 ||
+      (settings.internalApiPath &&
+        adjustedUrl.indexOf(settings.internalApiPath) !== -1) ||
+      adjustedUrl.indexOf(settings.apiPath) !== -1 ||
```

Comment:
It is not necessary to test `settings.publicURL`, `settings.internalApiPath` and `settings.internalApiPath` here, as they have been removed in the `adjustedUrl` assignment.